### PR TITLE
Feature/mlibz 2499 missing scope id mic login

### PR DESCRIPTION
**Pull request by vinaygahlawat**
#### Description
With MIC v3, **_"scope=openid"_** has to be passed in as a parameter to the **_/oauth/auth_** endpoint. This was missing in the render URL in normal MIC flow.

#### Changes
Add **_"scope=openid"_** to the render URL passed back to the MIC delegate during normal MIC flow.

#### Tests
Unit test added to make sure **_"scope=openid"_** is present in the URL.

### CODE REVIEW

**Comment by heyzooi** [comment]
@vinaygahlawat i'm wondering why this PR is having 20 commits and so many file changes? shouldn't this change to be a simpler change?

File: Kinvey.Core/Auth/SQLCredential.cs [hunk 1]
```diff
@@ -36,6 +36,12 @@ public class SQLCredential
 	/// <value>The auth token.</value>
 	public string AuthToken { get; set; }
 
+	/// <summary>
+	/// Gets or sets the auth token.
+	/// </summary>
+	/// <value>The auth token.</value>
+	public byte[] SecAuthToken { get; set; }
```
Comment (heyzooi) [inline Kinvey.Core/Auth/SQLCredential.cs]: what's the difference between `AuthToken` and `SecAuthToken`?

File: Kinvey.Core/Auth/User.cs [hunk 2]
```diff
@@ -549,7 +549,7 @@ static public void LoginWithMIC(string redirectURI, KinveyMICDelegate<User> MICD
 
 			ct.ThrowIfCancellationRequested();
 
-			string myURLToRender = hostname + "oauth/auth?client_id=" + clientID + "&redirect_uri=" + redirectURI + "&response_type=code";
+			string myURLToRender = hostname + "oauth/auth?client_id=" + clientID + "&redirect_uri=" + redirectURI + "&response_type=code" + "&scope=openid";
```
Comment (heyzooi) [inline Kinvey.Core/Auth/User.cs]: this + `kinveyApiVersion = "4"` should be the real change right? if so, the PR should be touching a smaller number of files

File: Kinvey.Core/File/KinveyFileRequest.cs [hunk 3]
```diff
@@ -32,16 +33,18 @@ internal KinveyFileRequest (AbstractClient client, string requestMethod, string
 		#region KinveyFileRequest upload methods
 
 		internal async Task uploadFileAsync(FileMetaData metadata, byte[] input)
-		{
-			await uploadFileAsync (metadata, new ByteArrayContent (input));
-		}
+        {
```
Comment (heyzooi) [inline Kinvey.Core/File/KinveyFileRequest.cs]: what's the relation between `KinveyFileRequest` and `scope=openid`?

**Comment by vinaygahlawat** [comment]
@heyzooi Sorry about the confusion, this PR is pointing to the wrong base, which needs to be updated. I will close this PR and re-open, where you will see the much smaller diff. 😄

**Pull request closed without merging**